testing/internal/e2e/target/target_test.go

```diff
@@ -0,0 +1,260 @@
+package target_test
+
+import (
+	"context"
+	"encoding/json"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/boundary/api/hostcatalogs"
+	"github.com/hashicorp/boundary/api/hosts"
+	"github.com/hashicorp/boundary/api/hostsets"
+	"github.com/hashicorp/boundary/api/scopes"
+	"github.com/hashicorp/boundary/api/targets"
+	"github.com/hashicorp/boundary/testing/internal/e2e"
+	"github.com/hashicorp/boundary/testing/internal/e2e/boundary"
+	"github.com/kelseyhightower/envconfig"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type testVars struct {
+	BoundaryTargetIp         string `envconfig:"BOUNDARY_E2E_TARGET_IP"`
+	BoundaryTargetSshKeyPath string `envconfig:"BOUNDARY_E2E_SSH_KEY_PATH"`
+}
+
+func TestConnectTargetCli(t *testing.T) {
+	var testConfig testVars
+	require.NoError(t, envconfig.Process("", &testConfig))
+
+	output := boundary.AuthenticateCli(t)
+	require.NoError(t, output.Err, string(output.Stderr))
+
+	// Create an org
+	output = e2e.RunCommand([]string{
+		"boundary", "scopes", "create",
+		"-name", "e2e Automated Test Org",
+		"-format", "json",
+	})
+	require.NoError(t, output.Err, string(output.Stderr))
+	var newOrgResult scopes.ScopeCreateResult
+	err := json.Unmarshal(output.Stdout, &newOrgResult)
+	require.NoError(t, err)
+	newOrg := newOrgResult.Item
+	t.Cleanup(func() {
+		e2e.RunCommand([]string{"boundary", "scopes", "delete", "-id", newOrg.Id})
+	})
+	t.Logf("Created Org Id: %s", newOrg.Id)
+
+	// Create a project
+	output = e2e.RunCommand([]string{
+		"boundary", "scopes", "create",
+		"-scope-id", newOrg.Id,
+		"-name", "e2e Automated Test Project",
+		"-format", "json",
+	})
+	require.NoError(t, output.Err, string(output.Stderr))
+	var newProjectResult scopes.ScopeCreateResult
+	err = json.Unmarshal(output.Stdout, &newProjectResult)
+	require.NoError(t, err)
+	newProject := newProjectResult.Item
+	t.Cleanup(func() {
+		e2e.RunCommand([]string{"boundary", "scopes", "delete", "-id", newProject.Id})
+	})
+	t.Logf("Created Project Id: %s", newProject.Id)
```

Comment:
I give it 1 more of these tests before you want to write `createProject` etc helpers 😁.

Reply:
Agreed. Didn't want to abstract things too much yet. I imagine we'll see some more refactoring once we start work on the next set of tests.